src/renderer/shared/api/price/coingecko/CoingeckoAdapter.ts

```diff
@@ -0,0 +1,48 @@
+import { AssetId, Currency, PriceObject, PriceAdapter, PriceItem, PriceRange } from '../common/types';
+import { getCurrencyChangeTitle } from '../common/utils';
+import { COINGECKO_URL } from './consts';
+
+export class CoinGeckoAdapter implements PriceAdapter {
+  async getPrice(ids: AssetId[], currencies: Currency[], includeRateChange: boolean): Promise<PriceObject> {
+    const response = await fetch(`${COINGECKO_URL}/simple/price`, {
+      method: 'GET',
+      body: JSON.stringify({
+        ids: ids.join(','),
+        vs_currencies: currencies.join(','),
+        include_24hr_change: includeRateChange ? 'true' : 'false',
+      }),
+    });
+
+    const data = await response.json();
+
+    const result = ids.reduce<PriceObject>((result, assetId) => {
```

Comment:
Added minor readability

```suggestion
    return ids.reduce<PriceObject>((acc, assetId) => {
      acc[assetId] = currencies.reduce<Record<string, PriceItem>>((result, currency) => {
        result[currency] = {
          price: data[assetId][currency],
          change: data[assetId][getCurrencyChangeTitle(currency)],
        };

        return result;
      }, {});

      return acc;
    }, {});
```